tests/unittests/TraceEventsTest.cpp

```diff
@@ -567,7 +567,78 @@ TEST_P(TraceEventsTest, deviceManagerEvents) {
   auto &traceEvents = context.getTraceContext()->getTraceEvents();
 
   ASSERT_GT(traceEvents.size(), 0);
-  checkEventTimestamps(traceEvents);
+  // CompleteEvents are not necessarily monitonically increasing since they are
+  // added to the log when they end, not when they start.
+}
+
+TEST(TraceEventsTest, nestedScopedEvents) {
```

Comment:
Consider adding a test like this but with explicitly terminated trace blocks to make sure that that their ending times are taken when `end()` is called and not when they are destroyed. Might be useful to mix explicitly and implicitly ended blocks as well.